sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+    result.Name = node["name"].get<std::string>();
+    result.Type = node["type"].get<std::string>();
+    result.ErrorMessage = node["errorMessage"].get<std::string>();
+    return result;
+  }
+};
+
+struct SetAccessControlRecursiveResponse
+{
+  uint64_t DirectoriesSuccessful;
+  uint64_t FilesSuccessful;
+  uint64_t FailureCount;
+  std::vector<AclFailedEntry> FailedEntries;
+
+  static SetAccessControlRecursiveResponse CreateFromJson(const nlohmann::json& node)
+  {
+    SetAccessControlRecursiveResponse result;
+    result.DirectoriesSuccessful = std::stoull(node["directoriesSuccessful"].get<std::string>());
+    result.FilesSuccessful = std::stoull(node["filesSuccessful"].get<std::string>());
+    result.FailureCount = std::stoull(node["failureCount"].get<std::string>());
+    for (const auto& element : node["failedEntries"])
+    {
+      result.FailedEntries.emplace_back(AclFailedEntry::CreateFromJson(element));
+    }
+    return result;
+  }
+};
+
+struct Path
+{
+  std::string Name;
+  bool IsDirectory;
+  std::string LastModified;
+  std::string ETag;
+  uint64_t ContentLength;
+  std::string Owner;
+  std::string Group;
+  std::string Permissions;
+
+  static Path CreateFromJson(const nlohmann::json& node)
+  {
+    Path result;
+    result.Name = node["name"].get<std::string>();
+    result.IsDirectory = static_cast<bool>(stoi(node["isDirectory"].get<std::string>()));
+    result.LastModified = node["lastModified"].get<std::string>();
+    result.ETag = node["eTag"].get<std::string>();
+    result.ContentLength = std::stoull(node["contentLength"].get<std::string>());
+    result.Owner = node["owner"].get<std::string>();
+    result.Group = node["group"].get<std::string>();
+    result.Permissions = node["permissions"].get<std::string>();
+    return result;
+  }
+};
+
+struct PathList
+{
+  std::vector<Path> Paths;
+
+  static PathList CreateFromJson(const nlohmann::json& node)
+  {
+    PathList result;
+    for (const auto& element : node["paths"])
+    {
+      result.Paths.emplace_back(Path::CreateFromJson(element));
+    }
+    return result;
+  }
+};
+
+struct FileSystem
+{
+  std::string Name;
+  std::string LastModified;
+  std::string ETag;
+
+  static FileSystem CreateFromJson(const nlohmann::json& node)
+  {
+    FileSystem result;
+    result.Name = node["name"].get<std::string>();
+    result.LastModified = node["lastModified"].get<std::string>();
+    result.ETag = node["eTag"].get<std::string>();
+    return result;
+  }
+};
+
+struct FileSystemList
+{
+  std::vector<FileSystem> Filesystems;
+
+  static FileSystemList CreateFromJson(const nlohmann::json& node)
+  {
+    FileSystemList result;
+    for (const auto& element : node["filesystems"])
+    {
+      result.Filesystems.emplace_back(FileSystem::CreateFromJson(element));
+    }
+    return result;
+  }
+};
+
+struct StorageError
+{
+
+  // The service error response object.
+  struct Error
+  {
+    std::string Code;  // The service error code.
```

Comment:
What's the point of nesting and nesting it again and again? Just because swagger defines it so we have to accept it?

Reply:
Flattened other result and left `StorageError` as is for the moment. I think we will implement storage exception so this error type will not be used.